src/OrbitGl/App.cpp

```diff
@@ -1800,40 +1802,47 @@ OrbitApp::RetrieveModuleFromRemote(const std::string& module_file_path) {
     return it->second.future;
   }
 
-  orbit_base::Future<ErrorMessageOr<std::string>> check_file_on_remote =
-      thread_pool_->Schedule([process_manager = GetProcessManager(), module_file_path]() {
-        if (absl::GetFlag(FLAGS_instance_symbols_folder).empty()) {
-          return process_manager->FindDebugInfoFile(module_file_path, {});
-        }
-        return process_manager->FindDebugInfoFile(module_file_path,
-                                                  {absl::GetFlag(FLAGS_instance_symbols_folder)});
-      });
+  orbit_base::Future<ErrorMessageOr<NotFoundOr<std::filesystem::path>>> check_file_on_remote =
+      thread_pool_->Schedule(
+          [process_manager = GetProcessManager(),
+           module_file_path]() -> ErrorMessageOr<NotFoundOr<std::filesystem::path>> {
+            std::vector<std::string> additional_instance_folder;
+            if (!absl::GetFlag(FLAGS_instance_symbols_folder).empty()) {
+              additional_instance_folder.emplace_back(absl::GetFlag(FLAGS_instance_symbols_folder));
+            }
+            return process_manager->FindDebugInfoFile(module_file_path, additional_instance_folder);
+          });
 
   orbit_base::StopSource stop_source;
 
   auto download_file = [this, module_file_path, stop_token = stop_source.GetStopToken()](
-                           ErrorMessageOr<std::string> result) mutable
+                           const NotFoundOr<std::filesystem::path>& remote_search_outcome) mutable
       -> orbit_base::Future<ErrorMessageOr<CanceledOr<std::filesystem::path>>> {
-    if (result.has_error()) return {result.error()};
+    // As a first step, we treat the not found message as an error
```

Comment:
If it's a first step of a larger refactoring that will change this, consider a TODO for reference.